SteamKit2/SteamKit2/Steam/Handlers/SteamFriends/Callbacks.cs

```diff
@@ -493,6 +493,78 @@ internal FriendMsgEchoCallback( CMsgClientFriendMsgIncoming msg )
         }
 
         /// <summary>
+        /// <para>This callback is fired in response to receiving historical messages.</para> 
+        /// See also <seealso cref="RequestOfflineMessages"/> and
+        /// <seealso cref="RequestMessageHistory(SteamID)"/>.
+        /// </summary>
+        public sealed class FriendMsgHistoryCallback : CallbackMsg
+        {
+            /// <summary>
+            /// Gets the result of the request.
+            /// </summary>
+            public EResult Result { get; private set; }
+
+            /// <summary>
+            /// Gets the SteamID of the user with whom these messages were exchanged.
+            /// </summary>
+            public SteamID SteamID { get; private set; }
+
+            /// <summary>
+            /// The messages exchanged with the user.
+            /// Offline messages are marked by having set <see cref="FriendMessage.Unread"/> to <c>true</c>
+            /// </summary>
+            public ReadOnlyCollection<FriendMessage> Messages { get; private set; }
+
+            internal FriendMsgHistoryCallback( CMsgClientFSGetFriendMessageHistoryResponse msg )
+            {
+                Result = ( EResult )msg.success;
+
+                SteamID = msg.steamid;
+
+                var messages = msg.messages
+                 .Select(m => new FriendMessage(m))
+                 .ToList();
+
+                Messages = new ReadOnlyCollection<FriendMessage>(messages);
+            }
+
+            /// <summary>
+            /// Represents a single Message sent to or received from a friend
+            /// </summary>
+            public sealed class FriendMessage
+            {
+                /// <summary>
+                /// The SteamID of the User that wrote the message
+                /// </summary>
+                public SteamID SteamID { get; private set; }
+
+                /// <summary>
+                /// Whether or not the message has been read, i.e., is an offline message.
+                /// </summary>
+                public bool Unread { get; private set; }
+
+                /// <summary>
+                /// The actualy message
+                /// </summary>
+                public string Message { get; private set; }
+
+                /// <summary>
+                /// The time (in UTC) when the message was sent
+                /// </summary>
+                public DateTime Timestamp { get; private set; }
+
+                internal FriendMessage( CMsgClientFSGetFriendMessageHistoryResponse.FriendMessage msg )
+                {
+                    SteamID = new SteamID(msg.accountid);
```

Comment:
Good catch. I think this also won't construct correctly (it will call the ulong constructor). Valve's code has a similar problem, but they've got a private uint constructor to prevent calling the ulong one accidentally.